lib/command/src/Obelisk/Command/Run.hs

```diff
@@ -70,30 +75,64 @@ data CabalPackageInfo = CabalPackageInfo
 preprocessorIdentifier :: String
 preprocessorIdentifier = "__preprocessor-apply-packages"
 
-run :: MonadObelisk m => m ()
+profile
+  :: MonadObelisk m
+  => FilePath
+  -> m ()
+profile profileBaseName = withProjectRoot "." $ \root -> do
+  putLog Debug "Using profiled build of project."
+  liftIO $ createDirectoryIfMissing False $ takeDirectory profileBaseName
+  let -- Sane flags to enable by default, enable time profiling +
+      -- closure heap profiling.
+      rtsFlags = [ "+RTS", "-p", "-po" <> profileBaseName, "-hc", "-RTS" ]
+      nixBuildExpr = [hereLit|
+with (import ./. {});
+let
+  exeSource = obelisk.nixpkgs.writeText "ob-run" ''
+module Main where
+
+import Control.Exception
+import Reflex.Profiled
+import System.Environment
+
+import qualified Obelisk.Run
+import qualified Frontend
+import qualified Backend
+
+main :: IO ()
+main = do
+  args <- getArgs
+  let port = read $ args !! 0
+      assets = args !! 1
+      profileFile = (args !! 2) <> ".rprof"
+  Obelisk.Run.run port (Obelisk.Run.runServeAsset assets) Backend.backend Frontend.frontend `finally` writeProfilingData profileFile
+  '';
+in obelisk.nixpkgs.runCommand "ob-run" {
+     buildInputs = [ (profiled.ghc.ghcWithPackages (p: [ p.backend p.frontend])) ];
+} "ghc -x hs -prof -fno-prof-auto ${exeSource} -o $out"
+|]
+  exePath <- nixCmd $ NixCmd_Build $ def
+      & nixBuildConfig_outLink .~ OutLink_None
+      & nixCmdConfig_target .~ Target
+        { _target_path = Nothing
+        , _target_attr = Nothing
+        , _target_expr = Just nixBuildExpr }
+  assets <- findProjectAssets root
+  putLog Debug $ "Assets impurely loaded from: " <> assets
+  freePort <- getFreePort
+  (_, _, _, ph) <- createProcess_ "runProfExe" $ setCwd (Just root) $ setDelegateCtlc True $ proc exePath ([show freePort, T.unpack assets, profileBaseName] <> rtsFlags)
+  _ <- liftIO $ waitForProcess ph
```

Comment:
I think we have a wrapper around this that you should use instead.

Reply:
It looks like the wrappers (readProcessAndLogOutput & callProcessAndLogOutput) hide stdout from the user. The default ob run can provide some helpful information like what URL to visit. I guess we could do `readProcessAndLogOutput (Error, Error)` to show this, but that seems a little bit hacky. Using createProcess_ matches the behavior of `ob run` and `ob repl` with the nixShellWithPkgs function.

Reply:
I mean just for `waitForProcess`

Reply:
I see... I was wrong. We don't have a wrapper for just `waitForProcess`

Reply:
I don't see one for waitForProcess. There is something called withProcess which might be usable though? But it ends up supplying "CreatePipe" like readProcessAndLogOutput.